repositories/mentor_repository.js

```diff
@@ -0,0 +1,103 @@
+const { getKnex } = require("../db");
+const userRepo = require('./user_repository');
+const logger = require('../logger')(__filename);
+const { ROLE_NAMES } = require("../utils");
+
+async function getMentors(userId, status){
+  const knex = getKnex();
+  const allowedStatuses = await getEligibleMentorStatuses(userId);
+  const query = knex("mentors")
+    .join("users", "mentors.user_id", "users.id")
+    .select(
+      "mentors.id as mentor_id",
+      "mentors.user_id",
+      "mentors.about",
+      "mentors.status",
+      "mentors.contact",
+      "users.name",
+      "users.img as img_link",
+      "users.discord_nickname"
+    );
+  if (status) {
+    if (!allowedStatuses.includes(status)) {
+      throw new Error('Requested status not permitted');
+    }
+    query.where("mentors.status", status);
+  } else {
+    query.whereIn("mentors.status", allowedStatuses);
+  }
+  return await query;
+}
+
+async function mentorAlreadyExists(userId)
+{
+  const knex = getKnex(); 
+  const mentor = await knex("mentors")
+    .where("user_id", userId)
+    .first();
+
+  return !!mentor;
+}
+
+async function getMentorById(userId, mentorId){
+  const knex = getKnex();
+  userId = 1;
+  try
+  {
+    const allowedStatuses = await getEligibleMentorStatuses(userId);
+    const mentor = await knex("mentors")
+      .join("users", "mentors.user_id", "users.id")
+      .where("mentors.id", mentorId)
+      .whereIn("mentors.status", allowedStatuses)
+      .select(
+        "mentors.id as mentor_id",
+        "mentors.user_id",
+        "mentors.about",
+        "mentors.status",
+        "mentors.contact",
+        "users.name",
+        "users.img as img_link",
+        "users.discord_nickname"
+      ).first();
+
+    if (!mentor) {
+      throw new Error('Mentor not found or access denied');
```

Comment:
let's use custom errors as well + i feel like this check also could be moved to the mentors' service....